### PR TITLE
Re-Import Global, Namespace and Nodes

### LINKED ISSUE #97: Question: How I should export dashboard json
Hi,

I'm prepare #79 and I have some trouble to export the JSON file a grafana instance.

If I import a dashboard and export again without any modifications, I get a lot of changes:

For example this commit does not contain any changes, from a lot of changes of JSON level: https://github.com/jkroepke/grafana-dashboards-kubernetes/commit/706315bc38eb2a4aa3145073d1b50c52ad5900ef

Thats how I export the JSON

image

What the recommend way? If the mention approch is the correct one, would it be possible to import and export all dashboard the keep my PR clean as possible? Otherwise, I had tons of non related changes.

### PR DESCRIPTION
**Pull request by jkroepke**
# Pull Request

### :mag_right: What kind of change is it?

- chore

### :dart: What has been changed and why do we need it?

- Just an import and export of the dashboards json via export function.

### :heavy_check_mark: Which issue(s) this PR fixes?

- Fixes #97 

### :speech_balloon: Additional information?

- Reduces unnecessary diffs in a future PR

### CODE REVIEW

File: dashboards/k8s-views-namespaces.json [hunk 1]
```diff
@@ -9,30 +9,36 @@
...
+      "id": "stat",
+      "name": "Stat",
       "version": ""
     },
     {
       "type": "panel",
-      "id": "stat",
-      "name": "Stat",
+      "id": "timeseries",
+      "name": "Time series",
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply dashboards/k8s-views-namespaces.json:41]: No. I guess dependencies are just ordered now and in sort order, sort is before timeseries.

File: dashboards/k8s-views-namespaces.json [hunk 2]
```diff
@@ -62,6 +68,7 @@
   "editable": true,
   "fiscalYearStartMonth": 0,
   "graphTooltip": 1,
+  "id": null,
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply dashboards/k8s-views-namespaces.json:71]: it was part of the export

File: dashboards/k8s-views-global.json [hunk 3]
```diff
@@ -1,47 +1,4 @@
 {
-  "__inputs": [
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply dashboards/k8s-views-global.json:2]: You are right, sometime goes wrong here

File: dashboards/k8s-views-namespaces.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (dotdc) [reply dashboards/k8s-views-namespaces.json:41]: ok good to know :+1:

File: dashboards/k8s-views-global.json [hunk 4]
```diff
@@ -2868,8 +2890,8 @@
       {
         "current": {
           "selected": false,
-          "text": "",
-          "value": ""
+          "text": "Prometheus",
+          "value": "d9803217-435f-4ee4-8d45-0e2f08370325"
```
Comment (jkroepke) [inline dashboards/k8s-views-global.json:2894]:
```suggestion
          "text": "",
          "value": ""
```

**Comment by jkroepke** [comment]
@dotdc did you had a chance for an additional review?

**Comment by dotdc** [comment]
Hi @jkroepke,
Just had a look and there are still a few things to change, but I don't see any queries with Windows support.
Maybe it would be easier to make a dashboard, then review the dashboard api details on review?
It won't be a lot, and you already know how to fix most of it.
Let me know.

**Comment by jkroepke** [comment]
> but I don't see any queries with Windows support.

This is correct. This PR should not contain any functional changes.

While I export the JSON dashboard with the windows changes first time, I had so many changes not related to Windows.

Base on that I decide to setup a PR that contains only changes from Grafana, if I do a import and export of the JSON again.

Windows related changed will be introduce in a follow PR

File: dashboards/k8s-views-global.json [hunk 5]
```diff
@@ -2968,6 +2985,6 @@
   "timezone": "",
   "title": "Kubernetes / Views / Global",
   "uid": "k8s_views_global",
-  "version": 36,
+  "version": 38,
```
Comment (dotdc) [inline dashboards/k8s-views-global.json:2988]: +2 instead of +1

File: dashboards/k8s-views-global.json [hunk 6]
```diff
@@ -68,6 +68,7 @@
   "editable": true,
   "fiscalYearStartMonth": 0,
   "graphTooltip": 1,
+  "id": null,
```
Comment (dotdc) [inline dashboards/k8s-views-global.json:71]: remove this line

File: dashboards/k8s-views-namespaces.json (hunk 2 above)
Comment (dotdc) [inline dashboards/k8s-views-namespaces.json:71]: remove this line

File: dashboards/k8s-views-nodes.json [hunk 7]
```diff
@@ -74,6 +74,7 @@
   "editable": true,
   "fiscalYearStartMonth": 0,
   "graphTooltip": 1,
+  "id": null,
```
Comment (dotdc) [inline dashboards/k8s-views-nodes.json:77]: remove this line

**Review by dotdc — CHANGES_REQUESTED** [review]

**Comment by dotdc** [comment]
> > but I don't see any queries with Windows support.
> 
> This is correct. This PR should not contain any functional changes.
> 
> While I export the JSON dashboard with the windows changes first time, I had so many changes not related to Windows.
> 
> Base on that I decide to setup a PR that contains only changes from Grafana, if I do a import and export of the JSON again.
> 
> Windows related changed will be introduce in a follow PR

Sure, just made the review again. 
All the rest should be good.

When you'll make the real PR, I'll test the dashboards on my side, and we can fix issues (if any).

File: dashboards/k8s-views-global.json (hunk 5 above)
Comment (jkroepke) [reply dashboards/k8s-views-global.json:2988]:
```suggestion
  "version": 37,
```

File: dashboards/k8s-views-global.json (hunk 6 above)
Comment (jkroepke) [reply dashboards/k8s-views-global.json:71]:
```suggestion
```

File: dashboards/k8s-views-namespaces.json (hunk 2 above)
Comment (jkroepke) [reply dashboards/k8s-views-namespaces.json:71]:
```suggestion
```

**Comment by dotdc** [comment]
@jkroepke 
Also, this is just a test, right? 
I don't think we'll be merging this without any functional changes.

**Comment by jkroepke** [comment]
Changes applied as recommend.

**Review by dotdc — APPROVED** [review]

**Comment by dotdc** [comment]
@jkroepke 
Do you want to merge this? Or was it just a test for the export?

**Comment by jkroepke** [comment]
Yes, please merge, I will setup a second one.

**Comment by dotdc** [comment]
@jkroepke 
Done, also pushed a commit with a version bump for the namespaces dashboard

**Pull request merged**